packages/api-explorer/src/utils/path.ts

```diff
@@ -128,3 +132,35 @@ export const getSpecKey = (location: HLocation | Location): string | null => {
   }
   return match?.groups?.specKey || null
 }
+
+/**
+ * Gets the scene type of the current page
+ * @param path path of browser location
+ * @returns string representing the scene type
+ */
+export const getSceneType = (path: string) => {
+  const match = matchPath<{ tagType: string }>(path, {
+    path: '/:specKey/:tagType',
+  })
+  return match ? match!.params.tagType : ''
+}
+
+/**
+ * Confirms if filter is valid for the page scene type
+ * @param location browser location
+ * @param filter filter tag for page
+ */
+export const isValidFilter = (
+  location: HLocation | Location,
+  filter: string
+) => {
+  const sceneType = getSceneType(location.pathname)
+  if (!sceneType) return false
+  else if (!filter.localeCompare('all', 'en', { sensitivity: 'base' }))
+    return true
+  else if (sceneType === 'methods') {
+    return methodFilterOptions.test(filter)
+  } else {
+    return typeFilterOptions.test(filter)
+  }
```

Comment:
consider using a switch statement or having a variable with the result and one return statement.